Week9 - homework SQL/LibrariesManagement.sql

```diff
@@ -0,0 +1,298 @@
+CREATE DATABASE LibrariesManagement;
+
+USE LibrariesManagement;
+
+CREATE TABLE Libraries
+(
+    Id INT IDENTITY(1, 1) PRIMARY KEY,
+    Name VARCHAR(50),
+    StreetName VARCHAR(50),
+    City VARCHAR(50)
+)
+
+CREATE TABLE Librarians
+(
+    Id INT IDENTITY(1, 1) PRIMARY KEY,
+    Name VARCHAR(30),
+    HireDate Date,
+    IsOnHoliday VARCHAR(3),
+    LibraryId INT REFERENCES Libraries(Id)
+)
+
+CREATE TABLE Books
+(
+    Id INT IDENTITY(1, 1) PRIMARY KEY,
+    Title VARCHAR(50),
+    AuthorFirstName VARCHAR(30),
+    AuthorLastName VARCHAR(30),
+    Type VARCHAR(10), -- crime, fantasy, horror, adventure
+)
+
+ALTER TABLE Books
+ADD AuthorEmail VARCHAR(50);
+
+CREATE TABLE Books_Libraries
+(
+    Id INT IDENTITY(1, 1) PRIMARY KEY,
+    BookId INT REFERENCES Books(Id),
+    LibraryId INT REFERENCES Libraries(Id)
+)
+
+CREATE TABLE Sales
+(
+    Id INT IDENTITY(1, 1) PRIMARY KEY,
+    NumberOfCopies INT,
+    LibraryId INT REFERENCES Libraries(Id),
+    LibrarianId INT REFERENCES Librarians(Id)
+)
+
+ALTER TABLE [dbo].[Sales] DROP CONSTRAINT [FK__Sales__Librarian__4316F928]  -- couldn't drop column Librarians without doing this
+ALTER TABLE [dbo].[Sales] DROP CONSTRAINT [FK__Sales__LibraryId__4222D4EF]
+
+ALTER TABLE Sales
+DROP COLUMN LibrarianId;
+ALTER TABLE Sales
+DROP COLUMN LibraryId;
+ALTER TABLE Sales
+ADD BookLibraryId INT REFERENCES Books_Libraries(Id)
+
+CREATE TABLE Books_Sales
+(
+    Id INT IDENTITY(1, 1) PRIMARY KEY,
+    BookId INT REFERENCES Books(Id),
+    SalesId INT REFERENCES Sales(Id)
+)
+
+--exercise1
+-- 1st improvement: inserting data to tables
```

Comment:
Not quite an improvement because otherwise you wouldn't be able to test your work. :)